hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -197,6 +232,33 @@ public LocalTopicStats getLocalTopicStats() {
         throw new UnsupportedOperationException("Locality is ambiguous for client!");
     }
 
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+
+        Collection<Data> dataCollection = objectToDataCollection(messages, getSerializationService());
+        ClientMessage request = TopicPublishAllCodec.encodeRequest(name, dataCollection);
+        invoke(request);
```

Comment:
I believe this should be called with `invokeOnPartition`. Also async ones should have `partitionId` as the fourth parameter. 
new ClientInvocation(getClient(), request, getName(), `partitionId`).invoke();

This is to guarantee ordering behaviour. See following for details. 
https://docs.hazelcast.org/docs/4.0.1/manual/html-single/index.html#understanding-topic-behavior

We can add tests as following for new methods to verify the behaviour.
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/topic/TopicTest.java#L141
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/topic/TopicTest.java#L220

It seems tests are missing on the client side. Don't forget to add them for client side as well ? 
Easiest could be to make ClientTopicTest to extend TopicTest. We don't have to, but we are using this pattern to avoid writing tests twice when it fits. See if these tests are suitable. Here is an example:
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueueBasicLocalTest.java#L29